pkg/controller/toolchainconfig/configuration.go

```diff
@@ -0,0 +1,61 @@
+package toolchainconfig
+
+import "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+type ToolchainConfig struct {
+	cfg *v1alpha1.ToolchainConfigSpec
+}
+
+func (c *ToolchainConfig) AutomaticApproval() *autoApproval {
```

Comment:
Maybe it's better to return the value here and all other places where we do not use pointers in the API?
I mean:
```suggestion
func (c *ToolchainConfig) AutomaticApproval() autoApproval {
```
Here and in all the places. Just to avoid confusion. We will never return a nil here (and we should not).

Reply:
Good point, done in https://github.com/codeready-toolchain/host-operator/pull/438/commits/90af2f9b7c30cd88fe31d833c812ade8e60a2315